src/main/java/com/scorelab/openads/receiver/PcapReceiver.java

```diff
@@ -0,0 +1,177 @@
+	
+	public PcapReceiver(int device, boolean promiscuousMode, int timeoutMillis, int snaplength, long maxpackets) {
+		// TODO Auto-generated constructor stub
+		super(StorageLevel.MEMORY_AND_DISK_2());
+	    this.device = device;
+	    this.promiscuousMode = promiscuousMode;
+	    this.timeoutMillis = timeoutMillis;
+	    this.snaplength = snaplength;
+	}
+
+	@Override
+	public void onStart() {
+		// TODO Auto-generated method stub
+		new Thread(){
+			public void run(){
+				receive();
+			}
+		}.start();
+	}
+
+	private void receive() {
+		// TODO Auto-generated method stub
+		Gson gson = new Gson();
+		PcapHandle handle = null;
+		try {
+			handle = Pcaps.findAllDevs().get(this.device)
+			           .openLive(this.snaplength, 
+			        		this.promiscuousMode?PromiscuousMode.PROMISCUOUS:PromiscuousMode.NONPROMISCUOUS, 
+			        		this.timeoutMillis);
+		}catch (PcapNativeException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+			System.exit(0);
+		}
+		
+		int num = 0;
+		while (!isStopped()) {
+			Packet packet = null;
+			try {
+				packet = handle.getNextPacket();
+				if (packet == null) {
+					continue;
+				}else {
+					System.out.println(handle.getTimestamp());
+					System.out.println(packet);
+					num++;
+			        if ( maxpackets!= -1 && num >= maxpackets) {
+			          break;
+			        }
+			        store(gson.toJson(packet));
+				}
+			}catch(NotOpenException e) {
+				// TODO Auto-generated catch block
+				e.printStackTrace();
+			}finally{
+				handle.close();
+			}
+	    }
+		
+	}
+
+	@Override
+	public void onStop() {
+		// TODO Auto-generated method stub
+		
+	}
+	
+	public static void main(String[] args){
+		if(args.length < 2){
+			System.err.println("Usage: JavaCustomReceiver <hostname> <port>");
+			System.exit(1);
+		}
+		
+		//Set Logger level
+		Logger.getRootLogger().setLevel(Level.WARN);
+		
+		//Setup configuration, with 1 second batch size
+		SparkConf sparkConf = new SparkConf().setAppName("PcapReceiver");
+		JavaStreamingContext jsc = new JavaStreamingContext(sparkConf, new Duration(1000));
+		
+		//Create Argument Parser
+		Options options = new Options();
+		options.addOption("PromiscuousMode", "Promiscuous Mode, True of Flase");
+		options.addOption("timeoutMillis", "TimeoutMillis, eg. 100 ms");
+		options.addOption("snaplength", "snap length, eg. 65536 bytes");
+		options.addOption("maxpackets", "Receiving maximum packets number");
+		options.addOption("device", "PCAP monitoring device number, [0-9]");
+		CommandLineParser parser = new DefaultParser();
+		CommandLine cmd = null;
+		try {
+			cmd = parser.parse( options, args);
+		} catch (ParseException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+			System.exit(0);
+		}
+		
+		//check whether the argument provides whole information
+		if(!cmd.hasOption("device")||!cmd.hasOption("PromiscuousMode")||!cmd.hasOption("timeoutMillis")||!cmd.hasOption("snaplength")){
```

Comment:
It's better if you can use Java system variables to load parameters, or using a property file